go/types/map_test.go

```diff
@@ -547,6 +547,17 @@ func TestMapHas(t *testing.T) {
 	doTest(getTestRefToValueOrderMap, 2)
 }
 
+func TestMapRemoveMasksUnderlyingMap(t *testing.T) {
+	assert := assert.New(t)
+	vrw := newTestValueStore()
+
+	k := String("foo")
+	me := NewMap(vrw, k, String("bar")).Edit()
+	me.Remove(k)
+	assert.False(me.Has(k))
```

Comment:
Can you also add 2 more cases:

 1. Add a key-value-pair to the map editor and remove that.
 2. Add a key-value-pair where the value is another editor and remove that.